server/src/distributed_engine/mod.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{storage_engine::StorageEngine, EngineError};
+
+pub struct DistributedEngine<Storage: StorageEngine> {
+    pub local_storage: Storage,
+}
+
+impl<Storage> DistributedEngine<Storage>
+where
+    Storage: StorageEngine,
+{
+    pub fn new(local_storage: Storage) -> Self {
+        Self { local_storage }
+    }
+
+    pub async fn create_dir(&self, path: String) -> Result<(), EngineError> {
+        // a temporary implementation
```

Comment:
I think it's necessary to implement `create_dir()`, `delete_dir()`...ect in distributed_engine. And we also need to implement assistant functions like `add_sub_dir()`.